fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/SkewShuffleJoinEliminationRule.java

```diff
@@ -0,0 +1,514 @@
+                    .setInputs(rightExchangeOptExp.getInputs())
+                    .setLogicalProperty(rightExchangeOptExp.getLogicalProperty())
+                    .setStatistics(rightExchangeOptExp.getStatistics())
+                    .setCost(rightExchangeOptExp.getCost()).build();
+
+            Pair<ScalarOperator, ScalarOperator> leftTablePredicates =
+                    generateInAndNotInPredicate(leftSkewColumn, skewValues);
+            Pair<ScalarOperator, ScalarOperator> rightTablePredicates =
+                    generateInAndNotInPredicate(rightSkewColumn, skewValues);
+
+            Map<ColumnRefOperator, ColumnRefOperator> leftSplitOutputColumnRefMap =
+                    generateOutputColumnRefMap(
+                            leftExchangeOptExp.getOutputColumns().getColumnRefOperators(columnRefFactory));
+            Map<ColumnRefOperator, ColumnRefOperator> rightSplitOutputColumnRefMap =
+                    generateOutputColumnRefMap(
+                            rightExchangeOptExp.getOutputColumns().getColumnRefOperators(columnRefFactory));
+
+            PhysicalSplitConsumeOperator leftSplitConsumerOptForShuffleJoin =
+                    new PhysicalSplitConsumeOperator(leftSplitProduceOperator.getSplitId(),
+                            leftTablePredicates.second,
+                            leftExchangeOp.getDistributionSpec(), leftSplitOutputColumnRefMap);
+
+            PhysicalSplitConsumeOperator leftSplitConsumerOptForBroadcastJoin =
+                    new PhysicalSplitConsumeOperator(leftSplitProduceOperator.getSplitId(), leftTablePredicates.first,
+                            new RoundRobinDistributionSpec(), leftSplitOutputColumnRefMap);
+
+            PhysicalSplitConsumeOperator rightSplitConsumerOptForShuffleJoin =
+                    new PhysicalSplitConsumeOperator(rightSplitProduceOperator.getSplitId(),
+                            rightTablePredicates.second,
+                            rightExchangeOpOfOriginalShuffleJoin.getDistributionSpec(), rightSplitOutputColumnRefMap);
+
+            PhysicalSplitConsumeOperator rightSplitConsumerOptForBroadcastJoin =
+                    new PhysicalSplitConsumeOperator(rightSplitProduceOperator.getSplitId(),
+                            rightTablePredicates.first,
+                            DistributionSpec.createReplicatedDistributionSpec(), rightSplitOutputColumnRefMap);
+
+            Projection projectionOnJoin = originalShuffleJoinOperator.getProjection();
+
+            PhysicalHashJoinOperator newShuffleJoinOpt = new PhysicalHashJoinOperator(
+                    originalShuffleJoinOperator.getJoinType(), originalShuffleJoinOperator.getOnPredicate(),
+                    originalShuffleJoinOperator.getJoinHint(), originalShuffleJoinOperator.getLimit(),
+                    originalShuffleJoinOperator.getPredicate(), projectionOnJoin,
+                    originalShuffleJoinOperator.getSkewColumn(), originalShuffleJoinOperator.getSkewValues());
```

Comment:
> why use `originalShuffleJoinOperator.getSkewValues()` rather than `skewValues ` ?

actually skewValus and skewColumn only used in this rule, so it's ok to pass originalShuffleJoinOperator.getSkewValues(), but i will modify it for readability